runtime/io/io.cpp

```diff
@@ -0,0 +1,389 @@
+#include <gmp.h>
+#include <cstring>
+#include <stdexcept>
+
+#include "fcntl.h"
+#include "unistd.h"
+#include "runtime/alloc.h"
+#include "runtime/header.h"
+
+extern "C" {
+
+#define KCHAR char
+#define ERRTAG(err) "Lbl'hash'" #err
+
+  mpz_ptr move_int(mpz_t);
+
+  static block dotK = {getBlockHeaderForSymbol((((uint64_t)getTagForSymbolName("dotk")) << 32) | 1), {}};
+  static blockheader kseqHeader = {getBlockHeaderForSymbol((uint64_t)getTagForSymbolName("kseq{}"))};
+
+  block * block_errno() {
+    block * retBlock = static_cast<block *>(koreAlloc(sizeof(block)));
+    switch (errno) {
+    case EOF:
+      retBlock->h = getBlockHeaderForSymbol((((uint64_t)getTagForSymbolName(ERRTAG(EOF))) << 32) | 1);
```

Comment:
these are all 0 argument symbols so you shouldn't be calling getBlockHeaderForSymbol, you should just take the ored value and cast it to a pointer.